### PR TITLE
Minor fixes for RepeatingGroup in non-standard modes

### LINKED ISSUE #545: Layout for repeating group with "filter" and "showAll"
### Description of the bug

When using **filter** and **showAll** for an ordinary group (not just a likert-group), it is possible to hide individual rows to produce a table-like structure. On every row I have a single field that indicates if it should be hidden or not, so filter works fine, but the layout is bad. Every "row" is shown as a rather big box. The box has a strange line in it when the window is wide.
image

When the window is smaller, the line disappears:
image

If it were possible to make the layout "tighter", more like a likert-group, it would start looking more like a table.

### Steps To Reproduce

Look at app https://altinn.studio/repos/xmrsa/ra0749-vare-01
on page 04_Omsetning to see the code behind the screenshots above. The checkboxes are on the two pages 02_Naering and 02_Binæring. There is also a similar case on page 08_IKT_Investeringer where the radiobuttons being answered with Yes in a likert group decides which repeating group is shown bellow (on the same page).
image

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by olemartinorg**
## Description

I was supposed to double-check that #545 was indeed fixed before closing it, but I saw that the label for the repeating group was rendered inside the `FullWidthWrapper` and thus looked awkwardly misplaced:

20240902_15h03m28s_grim

This fixes the issue, and grabs a percy snapshot to make sure mode:showAll looks good into the future as well.

I also refactored the way we render these modes in order to properly render the label for the group as a legend/fieldset, and while writing cypress tests for all the non-standard modes I found that you could get deadlocked inside the edit container while in `hideTable` mode when the save button was hidden.

## Related Issue(s)

- closes #545

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [ ] Unit test(s) have been added/updated
  - [x] Cypress E2E test(s) have been added/updated
  - [ ] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [ ] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [x] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been added/updated
  
  - [x] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [x] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

File: src/layout/RepeatingGroup/RepeatingGroupContainer.tsx [hunk 1]
```diff
@@ -57,10 +61,13 @@ export const RepeatingGroupContainer = forwardRef((_, ref: React.ForwardedRef<HT
...
       ref={ref}
     >
-      {(!edit?.mode ||
-        edit?.mode === 'showTable' ||
-        edit?.mode === 'onlyTable' ||
-        (edit?.mode === 'hideTable' && !isEditingAnyRow)) && <RepeatingGroupTable />}
+      {showTable && <RepeatingGroupTable />}
+      {!showTable && textResourceBindings?.title && (
+        <Label
+          renderLabelAs='span'
```
Comment (cammiida) [inline src/layout/RepeatingGroup/RepeatingGroupContainer.tsx:67]: Burde labelen her være en `fieldset`/`legend` siden den er for en tabell?

Comment (cammiida) [reply src/layout/RepeatingGroup/RepeatingGroupContainer.tsx:67]: Er ingen ekspert selv, men så vidt jeg har skjønt så skal `fieldset` med `legend` brukes når det er en label for flere elementer.
https://html.spec.whatwg.org/multipage/form-elements.html#the-fieldset-element

Comment (olemartinorg) [reply src/layout/RepeatingGroup/RepeatingGroupContainer.tsx:67]: Allright, jeg skal se om det kan ordnes uten veldig mye krøll. :raised_hands:

**Review by cammiida — APPROVED** [review]
Klarer fortsatt ikke reprodusere, men Percy får det jo riktig og koden ser bra ut, så er sikkert bare noe rart hos meg 😅

**Pull request merged**